lib/switchtec.c

```diff
@@ -517,10 +518,16 @@ int switchtec_status(struct switchtec_dev *dev,
 	if (ret)
 		return ret;
 
+	if (switchtec_is_gen5(dev))
+		avail_port_bitmap = 0x0F00FFFFFFFFFFFF;
+	else
+		avail_port_bitmap = 0x000FFFFFFFFFFFFF;
 
 	for (i = 0; i < SWITCHTEC_MAX_PORTS; i++) {
-		if ((ports[i].stk_id >> 4) > SWITCHTEC_MAX_STACKS)
+		if ((ports[i].stk_id >> 4) > SWITCHTEC_MAX_STACKS ||
+		    !((avail_port_bitmap >> i) & 1))
```

Comment:
Why is this even necessary? Why does the previous check not correctly filter out missing ports? What is the actual problem you are fixing?

On my system 'i' is not the port ID so this doesn't even seem correct.